src/pages/ToDosPage.tsx

```diff
@@ -0,0 +1,313 @@
+    } catch {
+      setErrorType(ErrorType.Add);
+
+      return;
+    } finally {
+      triggerRequest();
+    }
+  };
+
+  const handleRemoveTodo = (todoId: number) => {
+    if (!user) {
+      return;
+    }
+
+    clearError();
+    setEditingTodo(todoId);
+    deleteToDos(todoId);
+  };
+
+  const handleClearCompleted = async () => {
+    if (!user) {
+      return;
+    }
+
+    try {
+      const completedTodosFromServer = await getCompletedTODOs(user.id);
+      const idS = getIdsForChanges(completedTodosFromServer);
+
+      deleteToDos(idS);
+    } catch {
+      setErrorType(ErrorType.Unexpected);
+    } finally {
+      triggerRequest();
+    }
+  };
+
+  const handleUpdateTodoStatus = (todoId: number, status: boolean) => {
+    if (!user) {
+      return;
+    }
+
+    clearError();
+    setEditingTodo(todoId);
+    updateStatus(todoId, status);
+  };
+
+  const handleAllToggler = async (isButtonActive: boolean) => {
+    if (!user) {
+      return;
+    }
+
+    try {
+      const todosFromServer = await getTODOs(user.id);
+      const todosForUpdate = todosFromServer.filter(({ completed }) => (
+        isButtonActive ? completed : !completed
+      ));
+      const idS = getIdsForChanges(todosForUpdate);
+
+      updateStatus(idS, isButtonActive);
+    } catch {
+      setErrorType(ErrorType.Unexpected);
+    } finally {
+      triggerRequest();
+    }
+  };
+
+  const handleTitleChange = (todoId: number, newTitle: string) => {
+    clearError();
+    setEditingTodo(todoId);
+
+    if (newTitle) {
+      updateTitle(todoId, newTitle);
+    } else {
+      deleteToDos(todoId);
+    }
+  };
+
+  useEffect(() => {
+    setIsAdding(false);
+    setEditingToDoIds([]);
+  }, [visibleToDos]);
+
+  useEffect(() => {
+    if (!user) {
+      return;
+    }
+
+    const setTodos = async () => {
+      try {
+        const todosFromServer = await getTODOs(user.id);
+
+        getSelectedTodos(todosFromServer);
+        setHasCompleted(todosFromServer.some(({ completed }) => completed));
+        setActiveCount(
+          todosFromServer.filter(({ completed }) => !completed).length,
+        );
+      } catch {
+        setErrorType(ErrorType.Unexpected);
+      } finally {
+        setIsTemp(false);
+      }
+    };
+
+    setTodos();
+  }, [filterStatus, requestCount]);
+
+  useEffect(() => {
+    setTimeout(clearError, 3000);
+  }, [requestCount]);
+
+  return (
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <Header
+          newTodoTitle={newTodoTitle}
+          onTitleChange={setNewTodoTitle}
+          onToDoAdd={handleAddTodo}
+          isAdding={isAdding}
+          isButtonActive={!activeCount && hasCompleted}
+          onToggleAll={handleAllToggler}
+        />
+
+        <ToDoList
+          todos={visibleToDos}
+          isTemp={isTemp}
+          onRemove={handleRemoveTodo}
+          deletingToDoId={editingTodoIds}
```

Comment:
why prop name is `deletingToDoId` if you pass `editingTodoIds` and use it to know if Todo is in **editing** status
![image](https://user-images.githubusercontent.com/86923089/209939074-16f60b00-1cd9-4e1c-9b80-8cb0d0139694.png)

Reply:
the same) 
didn't notice about renaming when changed logic)🙈